### PR TITLE
Fix mac osx issues with java home and jce

### PR DESCRIPTION
**Pull request by taylor-benson**
-Set java home on macosx using /usr/libexec/java_home
-Find command should have ./ for path to search, works for nix and mac

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add labels to this Pull Request</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a <a href="https://github.com/sous-chefs/java/blob/master/CHANGELOG.md">CHANGELOG</a> entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by taylor-benson** [comment]
@tas50 @erichelgeson 
Not sure if I was suppose to bump the metadata or if you guys handle it during the merge. Basically this PR resolves java home being set on mac clients (test using chef-client -z as well as our selenium cluster here) as well as jce resource with the find command to be POSIX compatible by including the path to be search.

**Comment by taylor-benson** [comment]
Also I don't know how to add labels to the PR. First time lol.

**Comment by damacus** [comment]
Hey @taylor-benson the labels (if you can see them) are on the right hand side of the page above projects.

**Comment by damacus** [comment]
Can you add a unit test to this please?

If you need help adding a unit test, I'm happy to show you how! :)

**Comment by taylor-benson** [comment]
@damacus I couldn't add a label, but I see you did for me. Thanks.

As for a unit test. I have updated the spec accordingly. However I think may actually need integration testing, I'm unsure. As I wrote the test for jdk 6 (default attribute) and 11. However my jdk11 test wouldn't pass until I installed jdk 11 on my machine. My guess is b/c I'm using a ShellOut command to set the jdk. 

Initially I had the java_home set like "$(/usr/libexec/java_home -v 1.#{node['java]['jdk_version})" when below jdk 10. But due to other references like in the jce resource where there is a file join (java_home, jdk_version), it explicitly created a directory $(/...

I use to have this in my bash profile for java home on my mac
`export JAVA_HOME="$(/usr/libexec/java_home -v 1.8)"`

**Comment by taylor-benson** [comment]
Just as I suspected, unit test failed. Works on my machine lol.

I don't have a kitchen image for mac to actually integration test. I was just using chef-client -z to do my testing.

Let me rethink this to not use shell out so it can be unit tested as

**Comment by taylor-benson** [comment]
Ok I updated the code to not use shell out and updated the appropriate tests.

I have tested here locally that the install and certificate resource works. The only issue I see is the sysout during the add cert will show like `add certificate <certName> to keystore $(/usr/libexec/java_home -v 1.8)/jre/lib/security/cacerts` however the shell expansion works and the cert is installed.

As for the jce, looking into the homebrew on mac, java8 automatically handles the jce, and the java's that are higher than 8 are using openjdk which already has the unlimited policies. So for the jce resource, I can remove the path that I set (however it has no negative impact).

Let me know if you guys need anything else.

Cheers

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 This file is used to list changes made in each version of the Java cookbook.
 
+## 3.1.2 - (2018-11-19)
+
+-Set java home on macosx using /usr/libexec/java_home
```
Comment (josephholsten) [inline CHANGELOG.md]:
```suggestion
- Set java home on macosx using /usr/libexec/java_home
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
...
+-Find command should have ./ for path to search, works for nix and mac
```
Comment (josephholsten) [inline CHANGELOG.md]:
```suggestion
- Find command should have ./ for path to search, works for nix and mac
```

File: spec/set_attributes_from_version_spec.rb [hunk 3]
```diff
@@ -2,44 +2,65 @@
 
 describe 'java::set_attributes_from_version' do
   platforms = {
-    'centos-6.9' => {
+    'centos-6.9-jdk6' => {
```
Comment (josephholsten) [inline spec/set_attributes_from_version_spec.rb]: This isn't really the way we usually do things, platform is an attribute from ohai that we want to keep the way it is. 

Let's remove the `-jdk...`suffix from all these.
```suggestion
    'centos-6.9' => {
```

File: spec/set_attributes_from_version_spec.rb [hunk 4]
```diff
@@ -2,44 +2,65 @@
...
       'java_home' => nil,
+      'jdk_version' => '6',
+      'packages' => [],
+    },
+    'mac_os_x-10.14-jdk6' => {
+      'java_home' => '$(/usr/libexec/java_home -v 1.6)',
+      'jdk_version' => '6',
+      'packages' => [],
+    },
+    'mac_os_x-10.14-jdk11' => {
```
Comment (josephholsten) [inline spec/set_attributes_from_version_spec.rb]:
```suggestion
    'mac_os_x-10.14' => {
```

File: spec/set_attributes_from_version_spec.rb [hunk 5]
```diff
@@ -2,44 +2,65 @@
...
+      'jdk_version' => '6',
       'packages' => ['openjdk-6-jdk', 'openjdk-6-jre-headless'],
     },
-    'windows-2008R2' => {
+    'windows-2008R2-jdk6' => {
       'java_home' => nil,
+      'jdk_version' => '6',
+      'packages' => [],
+    },
+    'mac_os_x-10.14-jdk6' => {
```
Comment (josephholsten) [inline spec/set_attributes_from_version_spec.rb]:
```suggestion
    'mac_os_x-10.14' => {
```

File: spec/set_attributes_from_version_spec.rb [hunk 6]
```diff
@@ -2,44 +2,65 @@
...
       'packages' => ['openjdk-6-jdk', 'openjdk-6-jre-headless'],
     },
-    'ubuntu-14.04' => {
+    'ubuntu-14.04-jdk6' => {
       'java_home' => '/usr/lib/jvm/java-6-openjdk-amd64',
+      'jdk_version' => '6',
       'packages' => ['openjdk-6-jdk', 'openjdk-6-jre-headless'],
     },
-    'windows-2008R2' => {
+    'windows-2008R2-jdk6' => {
```
Comment (josephholsten) [inline spec/set_attributes_from_version_spec.rb]:
```suggestion
    'windows-2008R2' => {
```

File: spec/set_attributes_from_version_spec.rb [hunk 7]
```diff
@@ -2,44 +2,65 @@
...
       'packages' => ['openjdk6'],
     },
-    'debian-7.11' => {
+    'debian-7.11-jdk6' => {
       'java_home' => '/usr/lib/jvm/java-6-openjdk-amd64',
+      'jdk_version' => '6',
       'packages' => ['openjdk-6-jdk', 'openjdk-6-jre-headless'],
     },
-    'ubuntu-14.04' => {
+    'ubuntu-14.04-jdk6' => {
```
Comment (josephholsten) [inline spec/set_attributes_from_version_spec.rb]:
```suggestion
    'ubuntu-14.04' => {
```

File: spec/set_attributes_from_version_spec.rb [hunk 8]
```diff
@@ -2,44 +2,65 @@
...
       'packages' => ['java-1.6.0-openjdk', 'java-1.6.0-openjdk-devel'],
     },
-    'freebsd-10.3' => {
+    'freebsd-10.3-jdk6' => {
       'java_home' => '/usr/local/openjdk6',
+      'jdk_version' => '6',
       'packages' => ['openjdk6'],
     },
-    'debian-7.11' => {
+    'debian-7.11-jdk6' => {
```
Comment (josephholsten) [inline spec/set_attributes_from_version_spec.rb]:
```suggestion
    'debian-7.11' => {
```

File: spec/set_attributes_from_version_spec.rb [hunk 9]
```diff
@@ -2,44 +2,65 @@
...
       'packages' => ['java-1.6.0-openjdk', 'java-1.6.0-openjdk-devel'],
     },
-    'redhat-6.9' => {
+    'redhat-6.9-jdk6' => {
       'java_home' => '/usr/lib/jvm/java-1.6.0',
+      'jdk_version' => '6',
       'packages' => ['java-1.6.0-openjdk', 'java-1.6.0-openjdk-devel'],
     },
-    'freebsd-10.3' => {
+    'freebsd-10.3-jdk6' => {
```
Comment (josephholsten) [inline spec/set_attributes_from_version_spec.rb]:
```suggestion
    'freebsd-10.3' => {
```

File: spec/set_attributes_from_version_spec.rb [hunk 10]
```diff
@@ -2,44 +2,65 @@
...
 describe 'java::set_attributes_from_version' do
   platforms = {
-    'centos-6.9' => {
+    'centos-6.9-jdk6' => {
       'java_home' => '/usr/lib/jvm/java-1.6.0',
+      'jdk_version' => '6',
       'packages' => ['java-1.6.0-openjdk', 'java-1.6.0-openjdk-devel'],
     },
-    'redhat-6.9' => {
+    'redhat-6.9-jdk6' => {
```
Comment (josephholsten) [inline spec/set_attributes_from_version_spec.rb]:
```suggestion
    'redhat-6.9' => {
```

**Pull request merged**